### PR TITLE
[nRF Connect] Port lighting example

### LINKED ISSUE #2530: [nRF Connect] Port lighting example
#### Problem
As for now the lighting-app only supports nRF5 SDK platform and Linux port is coming: #2365. We would like to have the app available for nRF Connect SDK as well.

 #### Proposed Solution
Port the lighting app for nRF5 to nRF Connect SDK

### PR DESCRIPTION
**Pull request by Damian-Nordic**
#### Problem
As for now the lighting-app is only present on nRF5 SDK platform and Linux port is coming: #2365. We would like to have the app available for nRF Connect SDK as well.

 #### Summary of Changes
* Port all functionalities currently present in lighting-app/nrf5 to lighting-app/nrfconnect
* Extract LEDWidget to examples/platform/nrfconnect to reduce code duplication
* Handle a button triggering Thread joiner role (just like in the lock-app)
* A couple of minor fixes and improvements

Note that Server.cpp and DataModelHandler.cpp will likely also be made reusable after #2367 is merged.

fixes #2530

### CODE REVIEW

**Review by rwalker-apple — APPROVED** [review]

**Comment by Damian-Nordic** [comment]
@rwalker-apple FYI I had to update .restyled.yaml to fix restyled crashes on this PR: https://github.com/project-chip/connectedhomeip/pull/2531/commits/2f27afca2658fac75407d6cecf884e542555b312. After that the restyled job succeeded on my fork, but seems like it's still not working on the CHIP repo as the previous job has been stuck for about an hour by now: https://restyled.io/gh/project-chip/repos/connectedhomeip/jobs/370382. Do you know if there's any way to cancel tasks and push the queue forward?

File: examples/lighting-app/nrfconnect/README.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+### Using Docker container
+
+The easiest way to get started with the example is to use nRF Connect SDK Docker
+image for CHIP applications. Run the following commands to start a Docker
+container:
+
+        $ mkdir ~/nrfconnect
+        $ mkdir ~/connectedhomeip
+        $ docker run --rm -it --privileged -v /dev:/dev -v ~/nrfconnect:/var/ncs -v ~/connectedhomeip:/var/chip nordicsemi/nrfconnect-chip
```
Comment (mspang) [inline examples/lighting-app/nrfconnect/README.md:100]: I tried this yesterday and something the container did turned the /dev/null device on my host system into a regular file. I had to reboot to recover.

This is pretty risky - far worse than corrupting /dev/null could happen with root access to all devices. I'm not sure we should recommend this workflow.

**Review by mspang — APPROVED** [review]

File: .restyled.yaml [hunk 2]
```diff
@@ -43,6 +43,7 @@ exclude:
     - ".github/workflows/*" # https://github.com/restyled-io/restyler/issues/73
     - ".github/**/*" # https://github.com/restyled-io/restyler/issues/73
     - ".github/*" # https://github.com/restyled-io/restyler/issues/73
+    - "**/third_party/connectedhomeip" # restyler falls into an infinite loop due to symlinks to the root directory
```
Comment (rwalker-apple) [inline .restyled.yaml:46]: this should be unnecessary after a rebase

File: examples/lighting-app/nrfconnect/README.md (hunk 1 above)
Comment (Damian-Nordic) [reply examples/lighting-app/nrfconnect/README.md:100]: Interesting, it's never happened to me. I'll change the instruction to only expose serial devices.

**Comment by mspang** [comment]
@saurabhst @BroderickCarlin @jelderton @hawk248

**Review by jelderton — APPROVED** [review]

**Pull request merged**